aiida/cmdline/commands/cmd_export.py

```diff
@@ -186,86 +186,52 @@ def create(
     # version inside the function when needed.
     help='Archive format version to migrate to (defaults to latest version).',
 )
-def migrate(input_file, output_file, force, silent, in_place, archive_format, version):
-    # pylint: disable=too-many-locals,too-many-statements,too-many-branches
-    """Migrate an export archive to a more recent format version."""
-    from distutils.version import StrictVersion
-    import os
-    import shutil
-    import tarfile
-    import tempfile
-    import zipfile
-
-    from aiida.common import json
-    from aiida.common.folders import SandboxFolder
-    from aiida.tools.importexport import migration, extract_zip, extract_tar, ArchiveMigrationError, EXPORT_VERSION
+@click.option(
+    '--verbosity',
+    default='INFO',
+    type=click.Choice(['DEBUG', 'INFO', 'WARNING', 'CRITICAL']),
+    help='Control the verbosity of console logging'
+)
+def migrate(input_file, output_file, force, silent, in_place, archive_format, version, verbosity):
+    """Migrate an export archive to a more recent format version.
 
-    if version is None:
-        version = EXPORT_VERSION
+    .. deprecated:: 1.5.0
+        Support for the --silent flag, replaced by --verbosity
+
+    """
+    from aiida.common.log import override_log_formatter_context
+    from aiida.common.progress_reporter import set_progress_bar_tqdm
+    from aiida.tools.importexport import detect_archive_type, EXPORT_VERSION
+    from aiida.tools.importexport.archive.migrators import get_migrator, MIGRATE_LOGGER
+
+    if silent is True:
+        echo.echo_deprecated('the --silent option is deprecated, use --verbosity')
 
     if in_place:
         if output_file:
             echo.echo_critical('output file specified together with --in-place flag')
-        tempdir = tempfile.TemporaryDirectory()
-        output_file = os.path.join(tempdir.name, 'archive.aiida')
     elif not output_file:
         echo.echo_critical(
             'no output file specified. Please add --in-place flag if you would like to migrate in place.'
         )
 
-    if os.path.exists(output_file) and not force:
-        echo.echo_critical('the output file already exists')
+    if verbosity in ['DEBUG', 'INFO']:
+        set_progress_bar_tqdm(leave=(verbosity == 'DEBUG'))
+    MIGRATE_LOGGER.setLevel(verbosity)
 
-    with SandboxFolder(sandbox_in_repo=False) as folder:
+    if version is None:
+        version = EXPORT_VERSION
 
-        if zipfile.is_zipfile(input_file):
-            extract_zip(input_file, folder, silent=silent)
-        elif tarfile.is_tarfile(input_file):
-            extract_tar(input_file, folder, silent=silent)
-        else:
-            echo.echo_critical('invalid file format, expected either a zip archive or gzipped tarball')
+    migrator_cls = get_migrator(detect_archive_type(input_file))
+    migrator = migrator_cls(input_file)
```

Comment:
hehe, migrator is not a context manager?

I'm shocked :-D

Reply:
lol no I didn't see the need